sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosQueryRequestOptions.java

```diff
@@ -341,6 +344,16 @@ public CosmosQueryRequestOptions setCosmosEndToEndOperationLatencyPolicyConfig(C
         return this;
     }
 
+    /**
+     * Sets the {@link CosmosRetryStrategy} instance to be used for the request. This
+     * will override the behavior of the  {@link CosmosRetryStrategy} instance configured
+     * through {@link CosmosClientBuilder} instance.
+     * */
+    public CosmosQueryRequestOptions setCosmosRetryStrategy(CosmosRetryStrategy retryStrategy) {
```

Comment:
what if the retry strategy on the client is not null, and this is not set. What is the behavior on this?

Reply:
The goal is to default to the client-level behavior if not set here. Or another way to look at it is the setting here overrides client-level behavior.

Reply:
Keeping behavior on client-level only after offline discussion.